pkg/storage/util.go

```diff
@@ -56,6 +58,10 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	name := meta.Name()
+	if ok, msg := validation.ValidatePathSegmentName(name, false); !ok {
```

Comment:
My point is that your PathSegmentName is redundant with DNS1123Subdomain, which already exists.  Less code is more better